anndata/tests/helpers.py

```diff
@@ -153,26 +175,26 @@ def gen_typed_df_t2_size(m, n, index=None, columns=None) -> pd.DataFrame:
     return df
 
 
+default_key_types = (
+    sparse.csr_matrix,
+    np.ndarray,
+    pd.DataFrame,
```

Comment:
This is used for `layers` as well so that won't work.  Could refactor but I think what we have [here](https://github.com/scverse/anndata/pull/1028/files/83c6363a4767efbbbfbb110fc89e23e0549c3442#diff-1913caafc2716cc3f96cb7cf500214abb2ff07656987875eb03222f5e8b6e6c5R194-R196) is fine